include/xgboost/c_api.h

```diff
@@ -761,6 +761,39 @@ XGB_DLL int XGDMatrixNumRow(DMatrixHandle handle,
  */
 XGB_DLL int XGDMatrixNumCol(DMatrixHandle handle,
                             bst_ulong *out);
+
+
+/*!
+ * \brief Get number of valid values from DMatrix.
+ *
+ * \param handle the handle to the DMatrix
+ * \param out The output of number of non-missing values
+ *
+ * \return 0 when success, -1 when failure happens
+ */
+XGB_DLL int XGDMatrixNumNonMissing(DMatrixHandle handle, bst_ulong *out);
+
+/*!
+ * \brief Get the predictors from DMatrix as CSR matrix.  If this is a quantized DMatrix,
```

Comment:
Could it be useful to return quantised float values instead of integers?

We could return values such that `bst.predict(quantile_dmat) == bst.predict(xgb.DMatrix(quantile_dmat.get_data()))`

Reply:
Yes, we can. That's a good argument. I skipped restoring the values because it contains artificially created min and max values, which might be confusing.

Reply:
Changed the return value to be cut values.